mono/metadata/exception.c

```diff
@@ -45,13 +50,78 @@ static gpointer unhandled_exception_hook_data = NULL;
  *
  * \returns the initialized exception instance.
  */
+static MonoExceptionHandle
+mono_exception_new_by_name (MonoImage *image, const char *name_space, const char *name, MonoError *error)
+{
+	return mono_exception_new_by_name_domain (mono_domain_get (), image, name_space, name, error);
+}
+
 MonoException *
 mono_exception_from_name (MonoImage *image, const char *name_space,
 			  const char *name)
 {
 	return mono_exception_from_name_domain (mono_domain_get (), image, name_space, name);
 }
 
+/**
+ * mono_exception_from_name_domain:
+ * \param domain Domain where the return object will be created.
+ * \param image the Mono image where to look for the class
+ * \param name_space the namespace for the class
+ * \param name class name
+ *
+ * Creates an exception object of the given namespace/name class on
+ * the given domain.
+ *
+ * \returns the initialized exception instance.
+ */
+static MonoExceptionHandle
+mono_exception_new_by_name_domain (MonoDomain *domain, MonoImage *image,
+				 const char* name_space, const char *name, MonoError *error)
+{
+	HANDLE_FUNCTION_ENTER ()
+
+	MonoClass *klass;
+	MonoObjectHandle o = MONO_HANDLE_NEW (MonoObject, NULL);
+	MonoDomain *caller_domain = mono_domain_get ();
+
+	klass = mono_class_load_from_name (image, name_space, name);
+
+	mono_handle_assign_raw (o, mono_object_new_checked (domain, klass, error));
+	mono_error_assert_ok (error);
+	goto_if_nok (error, return_null);
```

Comment:
either the `mono_error_assert_ok` on the previous line or `goto_if_nok` on this line is redundant.

Reply:
I removed the assert. Caller can assert if it cannot tolerate errors.